flycheck.el

```diff
@@ -4335,6 +4348,23 @@ universal prefix arg, and only the id with normal prefix arg."
       (seq-do #'kill-new (reverse messages))
       (message (string-join messages "\n")))))
 
+(defun flycheck-explain-error-at-point ()
+  "Display an explanation for the first error at point."
+  (interactive)
+  (let* ((errors-at-point (flycheck-overlay-errors-at (point)))
+         (first-error (car errors-at-point))
+         (checker (and first-error (flycheck-error-checker first-error)))
+         (explainer (and checker
+                         (flycheck-checker-get checker 'error-explainer)))
+         (explanation (and explainer (funcall explainer first-error))))
```

Comment:
Even better, `-when-let*` will take care of the `(and ...` checks.  Though it seems `when-let` from `subr-x` does the same thing.  Should we prefer the dash version?